File: modules/wem-api/src/main/java/com/enonic/wem/api/aggregation/Aggregation.java
package com.enonic.wem.api.aggregation;

public abstract class Aggregation
{
    private final String name;

    public Aggregation( final Builder builder )
    {
        this.name = builder.name;
    }

    public String getName()
    {
        return name;
    }

    public static BucketAggregation.Builder bucketAggregation( final String name )
    {
        return new BucketAggregation.Builder( name );
    }


    public static class Builder<T extends Builder>
    {
        public Builder( final String name )
        {
            this.name = name;
        }

        private String name;
    }
}

File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/aggregation/Aggregation.ts

module api.aggregation {

    export class Aggregation {

        private name: string;

        constructor(name: string) {
            this.name = name;
        }

        public getName(): string {
            return this.name;
        }

        public static fromJsonArray(aggregationWrapperJsons: AggregationTypeWrapperJson[]) : Aggregation[] {

            var aggregations: Aggregation[] = [];

            aggregationWrapperJsons.forEach((aggregationJson: AggregationTypeWrapperJson) => {
                aggregations.push(AggregationFactory.createFromJson(aggregationJson));
            });

            return aggregations;
        }

    }
}
